package/harvester-os/files/usr/sbin/stream-disk

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+TARGET=/run/cos/target
+DATA_DISK_FSLABEL="HARV_LH_DEFAULT"
+DEFAULT_LH_PARTITION="6"
+
+update_boot_args()
+{
+      if [ -z "${HARVESTER_TTY}" ]; then
+          TTY=$(tty | sed 's!/dev/!!')
+      else
+          TTY=$HARVESTER_TTY
+      fi
+
+      if [ -e "/dev/${TTY%,*}" ] && [ "$TTY" != tty1 ] && [ "$TTY" != console ] && [ -n "$TTY" ]; then
+          sed -i "1s/.*/console_params=\"console=${TTY} console=tty1\"/g" ${TARGET}/etc/cos/bootargs.cfg
+      fi
+}
+
+echo "writing image to disk"
+curl --retry 5 --no-buffer -k ${HARVESTER_RAW_DISK_IMAGE_PATH} | dd of=${HARVESTER_DEVICE} bs=1M
```

Comment:
Where are the values of these two env variables set
- HARVESTER_RAW_DISK_IMAGE_PATH
- HARVESTER_DEVICE

Reply:
they are generated by call to: https://github.com/harvester/harvester-installer/blob/master/pkg/console/util.go#L482

This basically takes the harvester config and converts it to environment variables:
https://github.com/harvester/harvester-installer/blob/master/pkg/config/cos.go#L671

Reply:
`raw_disk_image_path` and `device` are fields in Harvester Config struct.

Reply:
Thanks